plutus-playground/plutus-playground-client/src/MainFrame.purs

```diff
@@ -307,18 +328,24 @@ replaceViewOnSuccess result source target = do
   when (isSuccess result && currentView == source)
     (assign _view target)
 
-currentEvaluation :: forall m. MonadState State m => SourceCode -> m Evaluation
+currentEvaluation :: forall m. MonadState State m => SourceCode -> m (Maybe Evaluation)
 currentEvaluation sourceCode = do
-  actions <- use _actions
-  let toPair :: MockWallet -> Tuple Wallet Int
-      toPair mockWallet =
-        view (_MockWallet <<< _wallet) mockWallet
-        /\
-        view (_MockWallet <<< _balance) mockWallet
   wallets <- map toPair <$> use _wallets
-  let program = toExpression <$> actions
-  let blockchain = []
-  pure $ Evaluation { wallets, program, sourceCode, blockchain }
+  simulation <- use _simulation
+  pure $ case simulation of
+    Nothing -> Nothing
+    Just (signature /\ actions) -> do
+      Just $ Evaluation { wallets
+                        , program: toExpression <$> actions
+                        , sourceCode
+                        , blockchain: []
+                        }
+  where
+    toPair :: MockWallet -> Tuple Wallet Int
```

Comment:
why does this mismatch occur, why do we need `Tuple Wallet Int` at all?

Reply:
Very good question... 🤔

Reply:
...ah, yes I remember. This is funny. It's because the backend uses a tuple, but I didn't like that so I used a record on the front. Such irony. I'll change the backend. :-D